sharding-core/sharding-core-common/src/main/java/org/apache/shardingsphere/core/metadata/TableMetaDataViolation.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.core.metadata;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import org.apache.shardingsphere.sql.parser.binder.metadata.table.TableMetaData;
+
+/**
+ * Violations holder for checking uniformed table meta data.
+ */
+@AllArgsConstructor
+@Getter
+class TableMetaDataViolation {
```

Comment:
How about use @RequiredArgsConstructor? It is better for final with fields.

Reply:
It couldn't be final fields, because the fields would be set value in that method.